validator/lib/jsonapi/validator/resource.rb

```diff
@@ -0,0 +1,123 @@
+require 'jsonapi/validator/document'
+require 'jsonapi/validator/relationship'
+
+module JSONAPI
+  module Validator
+    class Resource
```

Comment:
Well sinatra-param doesn't have an actionpack dependency, and there are others such as [json-schema](https://github.com/ruby-json-schema/json-schema).

My point is: this logic usually comes after deserialization, which means it's "one level up" and not jsonapi-specific. At that point there *should* be libraries already serving this purpose. Why do we need another one specific to jsonapi?

Once again, not against it at all just want to get a better understanding of the fuller picture.